index.bs

```diff
@@ -150,7 +196,20 @@ the following steps:
 1. Return |encodedOutput|.
 
 <h3 algorithm id="delivery-time">Establishing report delivery time</h3>
-TODO
+To <dfn>establish report delivery time</dfn> given an [=impression=] and a
+[=conversion/conversion time=], run the following steps:
+1. Let |timeDelta| be the difference between the [=conversion/conversion time=] and [=impression/impression time=].
+2. Let |expiryDelta| be the difference between the [=impression/expiry=] and the [=impression/impression time=]
+1. If |timeDelta| <= (2 days - 1 hour), return [=impression/impression time=] + 2 days
+1. If
+    - |expiryDelta| > (2 days - 1 hour) and
+    - |expiryDelta| < (7 days - 1 hour) and
+    - |timeDelta| <= |expiryDelta|
+
+
+    return the [=impression/expiry=] + 1 hour.
+1. If |timeDelta| <= (7 days - 1 hour), return [=impression/impression time=] + 7 days
```

Comment:
I'm having trouble understanding the conditions where this happens. The expiry is outside of the 2-day-to-7-day range, or the conversion happened after expiration? But what's the expiry for if not to prevent too-long conversion delays?